partcad/src/partcad/project.py

```
@@ -699,6 +711,63 @@ def get_part(self, part_name, func_params=None, quiet=False) -> Optional[Part]:
+
+        pc_logging.debug("Building %s:%s to resolve the part %s", self.name, owner, part_name)
+
+        # On a thread of its own: instantiating an assembly runs 'asyncio.run()',
+        # which raises if the calling thread already has a running event loop -
+        # and this is reached from both synchronous callers and coroutines.
+        failure = []
+
+        def build():
+            try:
+                asyncio.run(owning_assembly.do_instantiate())
+            except Exception as e:  # pylint: disable=broad-except
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)